connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/document/ExternalDocument.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.core.document;
+
+import io.camunda.connector.api.document.Document;
+import io.camunda.connector.api.document.DocumentLinkParameters;
+import io.camunda.connector.api.document.DocumentMetadata;
+import io.camunda.connector.api.document.DocumentReference;
+import io.camunda.connector.http.client.HttpClientService;
+import io.camunda.connector.http.client.model.HttpClientRequest;
+import io.camunda.connector.http.client.model.HttpClientResult;
+import io.camunda.connector.http.client.model.HttpMethod;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.OffsetDateTime;
+import java.util.Base64;
+import java.util.Map;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExternalDocument implements Document {
+
+  private final String url;
+  private final String name;
+  private transient DocumentMetadata metadata;
+  private final HttpClientRequest request;
+  private final HttpClientService httpClientService;
+  private HttpClientResult result = null;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ExternalDocument.class);
+
+  public ExternalDocument(String url, String name) {
```

Comment:
Idea: To decouple the `ExternalDocument` from the actual HTTP handling and to avoid creating a new HttpClientService for every document you could add a parameter like `Supplier<HttpClientResult>` to the constructor and do the rest outside of this class.